subiquity/client/controllers/ubuntu_advantage.py

```diff
@@ -51,6 +68,31 @@ def run_answers(self) -> None:
         if "token" in self.answers:
             self.done(self.answers["token"])
 
+    def check_token(self, token: str):
+        """ Asynchronously check the token passed as an argument. """
+        async def inner() -> None:
+            try:
+                svcs = await self.ua_interface.get_avail_services(token=token)
```

Comment:
Did you consider adding an API for this and doing it in the server? Because I don't think we want to reimplement this logic again in the desktop client.

Reply:
Thanks! I've carded this since it would indeed be beneficial but requires some more work than mere UI adjustments :)